functions/prepared-check-sync/access-arrangements-sql-util.js

```diff
@@ -0,0 +1,87 @@
+'use strict'
+
+const sqlService = require('less-tedious')
+
+const checkTable = '[check]'
+const pupilAccessArrangementsTable = '[pupilAccessArrangements]'
+const pupilTable = '[pupil]'
+const { TYPES } = require('tedious')
+const schema = '[mtc_admin]'
+const config = require('../config')
+sqlService.initialise(config)
+
+const sqlUtil = {}
+const accessArrangementCodes = {}
+
+/**
+ * Constants to describe the access arrangements codes
+ */
+sqlUtil.AACODES = Object.freeze({
+  AUDIBLE_SOUNDS: 'ATA',
+  COLOUR_CONTRAST: 'CCT',
+  FONT_SIZE: 'FTS',
+  INPUT_ASSISTANCE: 'ITA',
+  NEXT_BETWEEN_QUESTIONS: 'NBQ',
+  QUESTION_READER: 'QNR',
+  NUMPAD_REMOVAL: 'RON'
+})
+
+sqlUtil.sqlFindPupilAccessArrangementsByCheckCode = async function (checkCode) {
+  const sql = `
+  SELECT pAA.*, pfs.code AS pupilFontSizeCode, pcc.code AS pupilColourContrastCode
+  FROM ${schema}.${pupilAccessArrangementsTable} pAA
+  LEFT OUTER JOIN ${sqlService.adminSchema}.[pupilFontSizes] pfs
+    ON pAA.pupilFontSizes_id = pfs.id
+  LEFT OUTER JOIN ${sqlService.adminSchema}.[pupilColourContrasts] pcc
+    ON pAA.pupilColourContrasts_id = pcc.id
+  INNER JOIN ${schema}.${pupilTable} p
+    ON pAA.pupil_id = p.id
+  INNER JOIN ${schema}.${checkTable} chk
+    ON p.id = chk.pupil_id
+  WHERE chk.checkCode = @checkCode`
+  const params = [
+    {
+      name: 'checkCode',
+      value: checkCode,
+      type: TYPES.UniqueIdentifier
+    }
+  ]
+  const result = await sqlService.query(sql, params)
+  return result
+}
+
+/**
+ * Find access arrangement codes by ids
+ * @param {Array} ids
+ * @returns {Array}
+ */
+sqlUtil.sqlFindAccessArrangementsCodesWithIds = async function (ids) {
+  if (Object.keys(accessArrangementCodes).length === 0) {
+    // init
+    await init()
+  }
+  const result = []
+  Object.keys(accessArrangementCodes).forEach(code => {
```

Comment:
Would `Array.filter()` be a better fit here?

Reply:
@jon-shipley simplified with filter as suggested